homeassistant/components/sensor/brottsplatskartan.py

```diff
@@ -0,0 +1,136 @@
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['brottsplatskartan==0.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_AREA = 'area'
+DEFAULT_NAME = 'Brottsplatskartan'
+DOMAIN = 'brottsplatskartan'
+EVENT_INCIDENT = '{}_incident'.format(DOMAIN)
+SCAN_INTERVAL = timedelta(minutes=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Inclusive(CONF_LATITUDE, 'coordinates'): cv.latitude,
+        vol.Inclusive(CONF_LONGITUDE, 'coordinates'): cv.longitude,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_AREA, default=''): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Brottsplatskartan platform."""
+    import brottsplatskartan
+    import uuid
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    name = config.get(CONF_NAME)
+    area = config.get(CONF_AREA)
+
+    # Every Home Assistant instance should have their own unique
+    # app parameter: https://brottsplatskartan.se/sida/api
+    app = 'ha-' + str(uuid.getnode())
+
+    bpk = brottsplatskartan.BrottsplatsKartan(
+        app=app, area=area,
+        latitude=latitude, longitude=longitude
+    )
+
+    add_entities(
+        [BrottsplatskartanSensor(bpk, name)], True
+    )
+
+
+class BrottsplatskartanSensor(Entity):
+    """Representation of a Brottsplatskartan Sensor."""
+
+    def __init__(self, bpk, name):
+        """Initialize the Brottsplatskartan sensor."""
+        import brottsplatskartan
+        self._attributes = {ATTR_ATTRIBUTION: brottsplatskartan.ATTRIBUTION}
+        self._brottsplatskartan = bpk
+        self._name = name
+        self._previous_incidents = set()
+        self._starting_up = True
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    def _incident_event(self, incident):
+        """Fire if an event occurs."""
+        data = {
+            'description': incident.get('description'),
+            'external_source_link': incident.get('external_source_link'),
+            'location': incident.get('title_location'),
+            'timestamp': incident.get('pubdate_iso8601'),
+            'type': incident.get('title_type'),
+        }
+
+        if incident.get('lat') and incident.get('lng'):
+            data.update(
+                {
+                    ATTR_LATITUDE: incident.get('lat'),
+                    ATTR_LONGITUDE: incident.get('lng'),
+                }
+            )
+        self.hass.bus.fire(EVENT_INCIDENT, data)
```

Comment:
Entities shouldn't fire custom events. Entities already spawn state changed events. If we need to fire events it should be done from a new component.

Reply:
Ok, did not know that! Removed the whole section "_incident_event" method.